### DISCUSSION TITLE
Performance testing / benchmarking potential

### DISCUSSION THREAD

**Discussion by leafstrat** [realworld-apps/realworld#1513] [General]
I think this project has huge potential to be a benchmark test of sorts. Since the app is the same but the frameworks are being changed with each implementation, you could basically score each implementation based on performance speed;
- How fast pages generate when a user requests them
- How fast the server renders pages 

There would be other factors to take into account which effect performance and make measuring more difficult. Like where it's being hosted, for example fly.dev will be faster than vercel simply because with **fly.dev (free tier)** you have a "personal" server and **vercel (free tier)** is shared. But if the hosting were streamlined to one specific provider, it would be really interesting and helpful to compare all the frameworks.

Anyway just curious if something like this was already being worked on or given any thought. It is what I've been most interested in when opening up the different framework examples, checking how fast they compare to eachother in my browser's network tools.

Something that would've helped me researching this would be having a demo URL if available accessible directly [in the projects index.](https://codebase.show/projects/realworld) Also showing the URL for it, so I could for example open every project that is hosted with the same provider for performance comparison.

**Comment by geromegrignon** [comment]
Hi that's not the current direction of the project.

I can't claim each implementation uses the same optimization level for a real benchmark.

Going this way would require more sponsorship:
- the current traffic on the server is already over available free tiers.
- the database traffic is close to the free limit.

And it would require some collaboration with hosts providers as free tier do not allow to host that much applications for a benchmark.

**Reply by leafstrat** [reply]
Appreciate your reply and love the project.

**Reply by geromegrignon** [reply]
It might happen at some time however.
I was trying to focus on updating the specs for way too much time.
The new next step is to create brand new specs for another application example, bringing more modern real-world use cases not covered by the current project (websockets, security, accessibility, authorization...). It might allow some more options about the vision and the goals of this project.

**Reply by leafstrat** [reply]
The way I would implement this would be to simply encourage consistent use of a single project host (e.g., Vercel) for demos to facilitate comparison. Most of the demos were already using Vercel, but there were some outliers like heroku and fly.dev.

Though not mandatory, just suggesting it and highlighting its benefits - allowing for easy performance testing using tools like Google Lighthouse. This approach would address the common concern of web developers regarding the "best" framework. [0]

[SSR performance comparison of JS frameworks](https://www.enterspeed.com/blog/we-measured-the-ssr-performance-of-6-js-frameworks-heres-what-we-found) demonstrates that frameworks like Astro, SvelteKit, and Preact often outperform Next.js. Shifting focus in this direction could make the project more valuable for both learning and established developers.

[0] https://www.reddit.com/r/nextjs/comments/16secal/what_should_i_use_next_remix_or_astro/

Cheers, Gerome

**Reply by geromegrignon** [reply]
Ok, I thought it was about host provider competition. About the host provider, most of the demos use Netlify, not Vercel.

**Reply by geromegrignon** [reply]
By the way, you can already find performance testing here: https://github.com/krausest/js-framework-benchmark
